tasks/upload.py

```diff
@@ -383,17 +385,21 @@ async def run_async_within_lock(
                     upload_context.prepare_kwargs_for_retry(kwargs)
                     self.retry(countdown=retry_countdown, kwargs=kwargs)
 
-        try:
-            checkpoints = checkpoints_from_kwargs(UploadFlow, kwargs)
-            checkpoints.log(UploadFlow.PROCESSING_BEGIN)
-        except ValueError as e:
-            log.warning(f"CheckpointLogger failed to log/submit", extra=dict(error=e))
-
         repoid = upload_context.repoid
         commitid = upload_context.commitid
         report_type = upload_context.report_type
         report_code = upload_context.report_code
 
+        checkpoints = None
+        if report_type == ReportType.COVERAGE:
```

Comment:
some line above you used `ReportType.COVERAGE.value`.
Please make sure that this comparison is accurate.

Reply:
I see why this is confusing. In this line, we are using the `report_type` from the `UploadContext` which is the enum itself. In the line above that you are referring to, we are using the `report_type` passed as an arg to the task, which is a `str`